app/shared/compact-reservation-list/CompactReservationList.spec.js

```diff
@@ -66,4 +68,28 @@ describe('shared/compact-reservation-list/CompactReservationList', () => {
       expect(getWrapper().find('.compact-reservation-list-subtitle')).to.have.length(0);
     });
   });
+
+  describe('remove button', () => {
+    it('is rendered if reservation is in removableReservations', () => {
+      const reservations = [Reservation.build({ foo: 'bar' })];
+      const removableReservations = [
+        Reservation.build(),
+        Reservation.build(),
+      ];
+      const props = { reservations, removableReservations };
+      const buttons = getWrapper(props).find({ glyph: 'remove-circle' });
+      expect(buttons).to.have.length(2);
+    });
+
+    it('passes onRemoveClick func to onClick prop', () => {
+      const removableReservations = [Reservation.build()];
+      const props = { removableReservations };
+      const button = getWrapper(props).find({ glyph: 'remove-circle' });
+      expect(defaultProps.onRemoveClick.callCount).to.equal(0);
```

Comment:
- [ ] Whenever we are testing functions with mocks, let's try to create the mock on the test that is using it. Trouble with mocks in defaultProps is that they can be used or called in multiple tests which can break some tests if we test callCounts and such. 

In this case you can just create the `onRemoveClick` mock in this test and pass it as a prop.